tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
+
+		for (Int_t i = 0; i < fNofPlanes; i++)
+		{
+		fhTofsync[i]->Write();
+		fh_tofd_TotPm[i][0]->Write();
+		fh_tofd_TotPm[i][1]->Write();
+		fh_tofd_TotPm[i][2]->Write();
+		}
+	}
+	if (fParameter == 1)
+	{
+		// Determine time offset of the 2 PMTs of one paddle. This procedure
+		// assumes a sweep run in the middle of the ToF wall horizontally.
+		// Since all paddles are mounted vertically one can determine the offset.
+		// Half of the offset is added to PM1 and half to PM2.
+		LOG(info) << "Calling function calcOffset";
+		calcOffset();
+		// Determine ToT offset between top and bottom PMT
+		LOG(info) << "Calling function calcToTOffset";
+		calcToTOffset(fTofdTotLow, fTofdTotHigh);
+		// Determine sync offset between paddles
+		LOG(info) << "Calling function calcSync";
+		calcSync();
+		LOG(error) << "Call walk correction before next step!";
+
+	/*	for (Int_t i = 0; i < fNofPlanes; i++)
+		{
+			for (Int_t j = 0; j < fPaddlesPerPlane; j++)
+			{
+				auto par = fHitPar->GetModuleParAt(i + 1, j + 1);
+				Int_t binmax = fh1_tofsync[i][j]->GetMaximumBin();
+				auto tofsync = fh1_tofsync[i][j]->GetXaxis()->GetBinCenter(binmax);
+				par->SetTofSyncOffset(tofsync - fMeanTof);
+				LOG(info) << " Plane  " << i + 1 << " Bar " << j + 1 << " Tof-Sync  " << tofsync;
+				fh1_tofsync[i][j]->Write();
+			}
+		fhTofsync[i]->Write();
+		fh_tofd_TotPm[i][0]->Write();
+		fh_tofd_TotPm[i][1]->Write();
+		fh_tofd_TotPm[i][2]->Write();
+		}*/
+	}
+	else if(fParameter == 0)
+	{
+		// Calculate walk corrections
+		LOG(info) << "Calling function";
+		//calcwalk();
+
+
+	}
+	else if (fParameter == 2)
+	{
+		//Calculate ToF
+		LOG(info) << "Calculate ToF";
+		for (Int_t i = 0; i < fNofPlanes; i++)
+		{
+			for (Int_t j = 0; j < fPaddlesPerPlane; j++)
+			{
+				auto par = fHitPar->GetModuleParAt(i + 1, j + 1);
+				Int_t binmax = fh1_tofsync[i][j]->GetMaximumBin();
+				auto tofsync = fh1_tofsync[i][j]->GetXaxis()->GetBinCenter(binmax);
+				par->SetTofSyncOffset(tofsync - fMeanTof);
+				LOG(info) << " Plane  " << i + 1 << " Bar " << j + 1 << " Tof-Sync  " << tofsync;
+				fh1_tofsync[i][j]->Write();
+			}
+		fhTofsync[i]->Write();
+		fh_tofd_TotPm[i][0]->Write();
+		fh_tofd_TotPm[i][1]->Write();
+		fh_tofd_TotPm[i][2]->Write();
+		}
+		// Determine effective speed of light in [cm/s] for each paddle
+		LOG(info) << "Calling function";
+		calcVeff();
+		// Determine light attenuation lambda for each paddle
+		LOG(info) << "Calling function calcLambda";
+		calcLambda(fTofdTotLow, fTofdTotHigh);
+
+		for(int i=0; i<4; i++){
+			fhTofsync[i]->Write();
+			fhTdiff[i]->Write();
+		fh_tofd_TotPm[i][1]->Write();
+		fh_tofd_TotPm[i][0]->Write();
+		fh_tofd_TotPm[i][2]->Write();
+			for(int j=0;j<44;j++){
+				fhSqrtQvsPosToT[i][j]->Write();
+			}
+		}
+	}
+	else if (fParameter == 3)
+	{
+		// calculation of position dependend charge
+		if (fTofdSmiley)
+		{
+			
+		//	LOG(info) << "Calling function smiley";
+		//	Double_t para2[4];
```

Comment:
See previous comments about commented code

Reply:
Can change para2[4]; for (int i=0; i<4;i++) to para2[4]{};

Reply:
Have split step "3" into two, the commented lines are in step "-3". For now leave it to be ignored. Can be removed, once the split is accepted.